### PR TITLE
feat(performance): Add a configurable highWaterMark for performance. 

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Set the default to 16384.

@jrgm, this builds on #12 and should be reviewed afterwards.

r?

### CODE REVIEW

File: bin/csv-to-json.js [hunk 1]
```diff
@@ -26,15 +27,17 @@ if (! program.fxa || ! program.salesforce) {
   process.exit(1);
 }
 
+let highWaterMark = parseInt(program.highwater || 16384, 10);
```
Comment (jrgm) [inline bin/csv-to-json.js]: const

File: bin/json-to-sqs.js [hunk 2]
```diff
@@ -35,29 +36,32 @@ if (! program.input) {
   process.exit(1);
 }
 
+let highWaterMark = parseInt(program.highwater || 16384, 10);
```
Comment (jrgm) [inline bin/json-to-sqs.js]: const

File: bin/regenerate-testdata.sh [hunk 3]
```diff
@@ -6,7 +6,7 @@ SALESFORCE=./test_data/salesforce.csv
 SALESFORCE_SORTED=./test_data/salesforce_sorted.csv
 EXPECTED_COUNTS=./test_data/expected.json
 
-COUNT=4000
+COUNT=5000000
```
Comment (jrgm) [inline bin/regenerate-testdata.sh]: So, the travis test times out with 5M records. Maybe 50K?

Comment (shane-tomlinson) [reply bin/regenerate-testdata.sh]: Shit, I didn't mean 5M. I meant to revert this.

**Comment by shane-tomlinson** [comment]
@jrgm, can you re-review please! I set the branch protection a bit too strict and the original r+ was removed.

**Pull request merged**